Formula/flutter.rb

```diff
@@ -0,0 +1,27 @@
+class Flutter < Formula
+  desc "Google’s UI toolkit for mobile, web, and desktop from a single codebase"
+  homepage "https://flutter.dev"
+  url "https://storage.googleapis.com/flutter_infra/releases/stable/macos/flutter_macos_v1.9.1+hotfix.6-stable.zip"
```

Comment:
Homebrew core builds from source, or installs cross-platform binaries. See https://docs.brew.sh/Acceptable-Formulae

Reply:
Thank you for the code review
Corrected as pointed out, the proper way is to change the PR to homebrew cask, but I have found similar PR in the past.
And it has come to the point of sending PR to the homebrew formulae.
Please tell me which is the best method.
https://github.com/Homebrew/homebrew-cask/pull/63170

Reply:
If it doesn't build from source it can't be a formula in homebrew-core. If you want to submit it as a cask or make a formula in a private tap is up to you.

Reply:
> If it doesn't build from source it can't be a formula in homebrew-core.

Can’t it be, though? This submission doesn’t build it from source, but I see no indication it can’t be done.

Reply:
@fxcoudert and @SMillerDev
Please tell us how home brew's philosophy is to provide packages that are not gui and that are not built from source.

Reply:
I have no idea what you're talking about but you have to follow this rule https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae and all the others on the page to have your formula accepted into homebrew-core.

Reply:
Are there cases that are not accepted by both HomeBrew core and cask?  I want to know a solution for it in a way other than tap.

Reply:
Build the software from source

Reply:
Of course, you can [build the flutter engine yourself](https://github.com/flutter/flutter/wiki/Custom-Flutter-Engine-Embedding-in-AOT-Mode ). This is intended for those who want to customize, so a normal user will build a completely meaningless build.  If the homebrew team stipulates building from source, I will introduce that redundant flow into formulae.

Reply:
Yes, that's what's the rule that I linked to says.

Reply:
@fxcoudert and @SMillerDev and @vitorgalvao

I want to hear the final opinion of the homebrew team.
Does it mean that HomeBrew, even if it comes with pre-built binaries, must be rebuilt to be included in the package?  Also, if I modify it as a formulae to build the flutter engine, will it be merged?

Reply:
If something can't build from source it could be a cask. If it can it can be a formula in homebrew-core. For that to happen, the rules that were linked twice already must be followed, including but not limited to building the software from source. As for it getting merged, that depends on if you write a formula of sufficient quality to include it in homebrew-core. I can't give any guarantee on merging it before you start.

Reply:
"If something can't build from source it could be a cask."
This doesn't seem to be the case, and this seems to be rejected for CUI-only binaries.
In other words, HomeBrew has the disadvantage that it cannot be accepted by both core and cask, no matter how large the tool is.

Regarding the creation of formulae including redundant builds, we will consider modifying this PR.

Reply:
> "If something can't build from source it could be a cask."
This doesn't seem to be the case

It is the case, because it isn’t true that this can’t be built from source. Or it is true, that point hasn’t been made with a justification.

> HomeBrew potentially has the disadvantage of being unacceptable in both the core and the cask, regardless of the popularity of the tool that sent the PR.

Barring special situations, if a tool is popular enough and it can be built from source, it will be added to Homebrew when someone submits it. If no one does and there’s no reason not too, it wasn’t that popular to our user base.

Reply:
Thank you for this bikeshedding session. None of this applies to flutter, it's well known and popular and builds from source and there has been plenty of explanation what the requirements are for the formula. If you have a pull request that builds from source we can reconsider.